packages/pwa/webpack.config.js

```diff
@@ -66,13 +68,39 @@ module.exports = {
             PL_TERMS_OF_SERVICE: terms_of_service,
         }),
         new CleanWebpackPlugin(),
+        {
+            apply(compiler) {
+                compiler.hooks.compilation.tap("Store Built Files for CSP", (compilation) => {
+                    HtmlWebpackPlugin.getHooks(compilation).beforeEmit.tapAsync(
+                        "Store Built Files for CSP",
+                        (data, callback) => {
+                            const builtFilesForCsp = [];
+
+                            compilation.chunks.forEach((chunk) => {
+                                builtFilesForCsp.push(...chunk.files);
+                            });
+
+                            // Manually add the files in for the CSP meta tag
+                            data.html = data.html.replace(
+                                "script-src ",
+                                `script-src ${builtFilesForCsp.map((file) => `${pwaUrl}/${file}`).join(" ")} `
+                            );
+
+                            callback(null, data);
+                        }
+                    );
+
+                    return true;
+                });
+            },
+        },
         new HtmlWebpackPlugin({
             title: name,
             template: resolve(__dirname, "src/index.html"),
             meta: {
                 "Content-Security-Policy": {
                     "http-equiv": "Content-Security-Policy",
-                    content: `default-src 'self' ${serverUrl} https://api.pwnedpasswords.com blob:; style-src 'self' 'unsafe-inline'; object-src 'self' blob:; frame-src 'self'; img-src 'self' blob: data: https:;`,
+                    content: `default-src 'self' ${serverUrl}; base-uri 'self'; script-src blob:; connect-src ${serverUrl} https://api.pwnedpasswords.com; style-src 'self' 'unsafe-inline'; object-src 'self' blob:; frame-src 'self'; img-src 'self' blob: data: https:; manifest-src 'self'; worker-src ${pwaUrl}/sw.js;`,
```

Comment:
`'self'` needs to be removed from all of these directives. Otherwise anything can be loaded from the same origin which defeats the purpose of the whole exercise ;)

Reply:
When you do, don't forget to add `https://icons.duckduckgo.com/ip3/*` to `img-src` though, which is used for loading the favicons.

Reply:
Good point! :facepalm: